client/lib/upgrades/actions/domain-management.js

```diff
@@ -48,13 +47,7 @@ function setPrimaryDomain( siteId, domainName, onComplete = noop ) {
 			return onComplete( error, data );
 		}
 
-		sites.setSelectedSite( siteId );
-
-		Dispatcher.handleServerAction( {
-			type: 'FETCH_SITES'
-		} );
-
-		sites.once( 'change', () => {
+		requestSite( siteId )( dispatch ).then( () => {
```

Comment:
If the handler for `PRIMARY_DOMAIN_SET_COMPLETED` needs a full site object, what happens when the site request fails?